### PR TITLE
[RESTEASY-1629]

### PR DESCRIPTION
**Pull request by ronsigal**
UndertowJaxrsServer can handle context and initial parameters.

### CODE REVIEW

**Comment by ronsigal** [comment]
I could just commit this myself, but I just want someone to know about it.

File: server-adapters/resteasy-undertow/src/main/java/org/jboss/resteasy/plugins/server/undertow/UndertowJaxrsServer.java [hunk 1]
```diff
@@ -151,11 +154,34 @@ public UndertowJaxrsServer deploy(ResteasyDeployment deployment)
...
       DeploymentInfo builder = undertowDeployment(deployment);
       builder.setContextPath(contextPath);
       builder.setDeploymentName("Resteasy" + contextPath);
+      builder.setClassLoader(deployment.getApplication().getClass().getClassLoader());
+      if (contextParams != null)
+      {
+         for (Iterator<String> it = contextParams.keySet().iterator(); it.hasNext(); )
+         {
+            String key = it.next();
+            builder.addInitParameter(key, contextParams.get(key));
```
Comment (asoldano) [inline server-adapters/resteasy-undertow/src/main/java/org/jboss/resteasy/plugins/server/undertow/UndertowJaxrsServer.java]: For better performances, the best practice here would be:

for (Entry<String, String> e : contextParams.entrySet()) {
  builder.addInitParameter(e.getKey(), e.getValue());
}

which avoids accessing the map again using the key.

File: server-adapters/resteasy-undertow/src/main/java/org/jboss/resteasy/plugins/server/undertow/UndertowJaxrsServer.java [hunk 2]
```diff
@@ -151,11 +154,34 @@ public UndertowJaxrsServer deploy(ResteasyDeployment deployment)
...
+         }
+      }  
+      if (initParams != null)
+      {
+         ServletInfo servletInfo = builder.getServlets().get("ResteasyServlet");
+         for (Iterator<String> it = initParams.keySet().iterator(); it.hasNext(); )
+         {
+            String key = it.next();
+            servletInfo.addInitParam(key, initParams.get(key));
```
Comment (asoldano) [inline server-adapters/resteasy-undertow/src/main/java/org/jboss/resteasy/plugins/server/undertow/UndertowJaxrsServer.java]: Same as above :-)

File: server-adapters/resteasy-undertow/src/test/java/org/jboss/resteasy/test/undertow/UndertowParameterTest.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+
+/**
+ * @tpSubChapter XXE
+ * @tpChapter Integration tests
+ * @tpTestCaseDetails Regression test for RESTEASY-1103
```
Comment (asoldano) [inline server-adapters/resteasy-undertow/src/test/java/org/jboss/resteasy/test/undertow/UndertowParameterTest.java]: Just a minor thing, this is due to copy&paste and the test is not really for RESTEASY-1103, right?

File: server-adapters/resteasy-undertow/src/main/java/org/jboss/resteasy/plugins/server/undertow/UndertowJaxrsServer.java (hunk 1 above)
Comment (ronsigal) [inline server-adapters/resteasy-undertow/src/main/java/org/jboss/resteasy/plugins/server/undertow/UndertowJaxrsServer.java]: Ah, cool. Thanks.

File: server-adapters/resteasy-undertow/src/test/java/org/jboss/resteasy/test/undertow/UndertowParameterTest.java (hunk 3 above)
Comment (ronsigal) [inline server-adapters/resteasy-undertow/src/test/java/org/jboss/resteasy/test/undertow/UndertowParameterTest.java]: Oops, right. I don't know if those annotations are even relevant to tests outside of the testsuite module, so I removed them.

**Comment by ronsigal** [comment]
Requested changes made.

**Pull request merged**